### ISSUE TITLE
TestSession.test_http_disable_dh fails with 5.5.0

### ISSUE THREAD

**Issue by heirecka** [streamlink/streamlink#5332]
### Checklist

- [X] This is a bug report and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed bug reports](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22bug%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest stable release

### Description

Since 5.5.0 TestSession.test_http_disable_dh started to fail. The urllib3 version is 1.26.15, output below under debug log.

### Debug log

```text
================================================================================= FAILURES =================================================================================
_____________________________________________________________________ TestSession.test_http_disable_dh _____________________________________________________________________

self = <tests.test_session.TestSession testMethod=test_http_disable_dh>

    def test_http_disable_dh(self):
        session = self.subject(load_plugins=False)
        assert isinstance(session.http.adapters["https://"], HTTPAdapter)
        assert not isinstance(session.http.adapters["https://"], TLSNoDHAdapter)
    
>       session.set_option("http-disable-dh", True)

tests/test_session.py:404: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pep517_tests_streamlink/usr/x86_64-pc-linux-gnu/lib/python3.10/site-packages/streamlink/session.py:493: in set_option
    self.options.set(key, value)
pep517_tests_streamlink/usr/x86_64-pc-linux-gnu/lib/python3.10/site-packages/streamlink/options.py:54: in set
    method(self, normalized, value)
pep517_tests_streamlink/usr/x86_64-pc-linux-gnu/lib/python3.10/site-packages/streamlink/session.py:119: in _set_http_disable_dh
    adapter = TLSNoDHAdapter()
/usr/lib/python3.10/site-packages/requests/adapters.py:155: in __init__
    self.init_poolmanager(pool_connections, pool_maxsize, block=pool_block)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <streamlink.plugin.api.http_session.TLSNoDHAdapter object at 0x7f04fbe22d10>, args = (10, 10), kwargs = {'block': False}

    def init_poolmanager(self, *args, **kwargs):
>       ctx = urllib3.util.create_urllib3_context()
E       AttributeError: module 'urllib3.util' has no attribute 'create_urllib3_context'

pep517_tests_streamlink/usr/x86_64-pc-linux-gnu/lib/python3.10/site-packages/streamlink/plugin/api/http_session.py:191: AttributeError
========================================================================= short test summary info ==========================================================================
FAILED tests/test_session.py::TestSession::test_http_disable_dh - AttributeError: module 'urllib3.util' has no attribute 'create_urllib3_context'
=============================================================== 1 failed, 5642 passed, 37 skipped in 22.51s ================================================================
```

**Comment by bastimeyer** [comment]
Yep, my bad. Our CI tests don't run on urllib3 1.26.x, so this wasn't caught. I also force-pushed a couple of times in #5326 with changes to the SSLContext initialization and then apparently didn't locally run tests again with 1.26.15.

The `create_urllib3_context` method doesn't get re-exported in the `urllib3.util.__init__` module on <2.0.0. The docs don't mention it:
- https://urllib3.readthedocs.io/en/stable/advanced-usage.html#custom-ssl-contexts
- https://github.com/urllib3/urllib3/blob/1.26.0/src/urllib3/util/__init__.py
- https://github.com/urllib3/urllib3/blob/2.0.0/src/urllib3/util/__init__.py

**Comment by bastimeyer** [comment]
Thanks for the report

**Comment by bastimeyer** [comment]
Fixed in 5.5.1
https://github.com/streamlink/streamlink/releases/tag/5.5.1

**Comment by heirecka** [comment]
Thank you for the quick fix!
